test/conformance/api/shared/util.go

```diff
@@ -118,17 +118,18 @@ func checkResponses(t pkgTest.TLegacy, num, min int, domain string, expectedResp
 	//   WHERE body IN $expectedResponses
 	//   GROUP BY body
 	// )
-	for _, ar := range actualResponses {
+	for i, ar := range actualResponses {
 		if er := substrInList(ar, expectedResponses); er != "" {
 			counts[er]++
 		} else {
 			badCounts[ar]++
+			t.Logf("For domain %s: got unexpected response for request %d", domain, i)
 		}
 	}
 
-	// Verify that the total expected responses match the number of requests made.
+	// Print unexpected responses for debugging purposes
 	for badResponse, count := range badCounts {
-		t.Logf("Saw unexpected response %q %d times.", badResponse, count)
+		t.Logf("For domain %s: saw unexpected response %q %d times.", domain, badResponse, count)
```

Comment:
```suggestion
		t.Logf("domain %s: saw unexpected response %q %d times.", domain, badResponse, count)
```

nit, to be consistent with the log below.